EKF/control.cpp

```diff
@@ -1414,6 +1414,12 @@ void Ekf::controlMagFusion()
 				_time_last_movement = _imu_sample_delayed.time_us;
 			}
 
+			// Force 3D fusion after takeoff for a short time to make sure we reset the heading
+			if (height_achieved && !_height_achieved_once) {
+				_height_achieved_once = true;
```

Comment:
I'm thinking about the use case where the vehicle lands, disarms, arms and takes off again. We should follow the same 3D fusion reset on subsequent takeoffs, so this flag will need to be reset when the EKF detects the vehicle is on ground.